crates/providers/src/provider.rs

```diff
@@ -359,8 +368,24 @@ impl<T: Transport + Clone + Send + Sync> TempProvider for Provider<T> {
     }
 
     /// Execute a smart contract call with [CallRequest] without publishing a transaction.
-    async fn call(&self, tx: CallRequest, block: Option<BlockId>) -> TransportResult<Bytes> {
-        self.inner.prepare("eth_call", (tx, block.unwrap_or_default())).await
+    ///
+    /// # Note
+    ///
+    /// Not all client implementations support state overrides.
+    async fn call(
+        &self,
+        tx: CallRequest,
+        block: Option<BlockId>,
+        state: Option<StateOverride>,
+    ) -> TransportResult<Bytes> {
+        // todo: if we just set `None` for the 3rd parameter, will this be ok for clients who do not
+        // expect a third parameter? if not, should we split this into `call` and
+        // `call_with_overrides`?
```

Comment:
I think separating both is okay. Frankly, not many people use call with overrides and having to set none becomes a chore

Reply:
Separate IMO yep